### ISSUE TITLE
ldconfig: /usr/lib/wsl/lib/libcuda.so.1 is not a symbolic link

### ISSUE THREAD

**Issue by seudonam** [microsoft/WSL#11229]
### Windows Version

Microsoft Windows [ version 10.0.19044.4046]

### WSL Version

2.0.14.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.133.1-1

### Distro Version

Debian

### Repro Steps

ldconfig: /usr/lib/wsl/lib/libcuda.so.1 is not a symbolic link
Everytime installing package it hapened! Are you guys gonna keep this bug till world's end?

### Expected Behavior

ldconfig: /usr/lib/wsl/lib/libcuda.so.1 is not a symbolic link
Everytime installing package it hapened! Are you guys gonna keep this bug till world's end?

### Actual Behavior

ldconfig: /usr/lib/wsl/lib/libcuda.so.1 is not a symbolic link
Everytime installing package it hapened! Are you guys gonna keep this bug till world's end?

**Comment by elsaco** [comment]
@seudonam you might want to take this issue with `Team Green` because those libraries are part of the Nvidia driver and not WSL.

The `soname` is the same and `ldconfig` complains:

```
root@texas:~# objdump -p /usr/lib/wsl/lib/libcuda.so | grep -i soname
  SONAME               libcuda.so.1
root@texas:~# objdump -p /usr/lib/wsl/lib/libcuda.so.1 | grep -i soname
  SONAME               libcuda.so.1
root@texas:~# objdump -p /usr/lib/wsl/lib/libcuda.so.1.1 | grep -i soname
  SONAME               libcuda.so.1
root@texas:~# sha256sum /usr/lib/wsl/lib/libcuda.so
1ca50006c1fb01732c85a895ef9720833a35f485280233f329cf602adb1733ef  /usr/lib/wsl/lib/libcuda.so
root@texas:~# sha256sum /usr/lib/wsl/lib/libcuda.so.1
1ca50006c1fb01732c85a895ef9720833a35f485280233f329cf602adb1733ef  /usr/lib/wsl/lib/libcuda.so.1
root@texas:~# sha256sum /usr/lib/wsl/lib/libcuda.so.1.1
1ca50006c1fb01732c85a895ef9720833a35f485280233f329cf602adb1733ef  /usr/lib/wsl/lib/libcuda.so.1.1

```

`not a symbolic link` is a harmless warning. And with Nvidia's developers immersed in AI and the stock going up-and-up I doubt they'll care about this issue!

**Comment by liesauer** [comment]
ummm, my question is i am installing a flash wsl system, why this stupid warning coming up?

**Comment by Max-Sum** [comment]
This problem still bugs me. I did a hard research from pytorch/pytorch#73487 to here.
Is this a WSL2 problem or Nvidia problem?

**Comment by dqsh06** [comment]
解决/sbin/ldconfig.real: /usr/lib/wsl/lib/libcuda.so.1 is not a symbolic link的问题
意思是说 /usr/lib/wsl/lib/libcuda.so.1 不是一个符号链接
它们的关系应该是：
libcuda.so -> libcuda.so.1 -> libcuda.so.1.1
那么，到/usr/lib/wsl/lib 目录
用命令  
sudo mkdir backup    #用超级用户，建立backup目录
sudo mv libcuda.so ./backup
sudo mv libcuda.so.1 ./backup
sudo ln -s libcuda.so.1.1 libcuda.so.1   # 建立超链接 libcuda.so.1 -> libcuda.so.1.1
sudo ln -s libcuda.so.1 libcuda.so       # 建立超链接 libcuda.so -> libcuda.so.1

**Comment by dcasota** [comment]
I agree with @dqsh06. The issue is mentioned at https://forums.developer.nvidia.com/t/wsl2-libcuda-so-and-libcuda-so-1-should-be-symlink/236301, and a fix, too.
```
cd /usr/lib/wsl/lib
rm libcuda.so libcuda.so.1
ln -s libcuda.so.1.1 libcuda.so.1
ln -s libcuda.so.1 libcuda.so
/sbin/ldconfig
```

**Comment by PCrazywang** [comment]
/sbin/ldconfig.real: Can't link /usr/lib/wsl/lib/libnvoptix_loader.so.1 to libnvoptix.so.1
